core/src/loader.rs

```diff
@@ -130,7 +154,8 @@ pub fn load_associated_token_account(
 /// # Returns
 /// * `Result<(), ProgramError>` - The result of the operation
 pub fn load_token_mint(info: &AccountInfo) -> Result<(), ProgramError> {
```

Comment:
also token22 for this too

Reply:
added `load_token_2022_program` for checking token22 program_id,